webapp/channels/src/packages/mattermost-redux/src/actions/threads.ts

```diff
@@ -391,6 +392,12 @@ export function setThreadFollow(userId: string, teamId: string, threadId: string
             dispatch(logError(error));
             return {error};
         }
+
+        // As a short term fix for https://mattermost.atlassian.net/browse/MM-62113, we will fetch
+        // the users team unreads after following or unfollowing a thread. This will ensure that the unreads are
+        // updated correctly in the case where the user is following or unfollowing a thread that has unread messages.
+        dispatch(getMyTeamUnreads(isCollapsedThreadsEnabled(getState())));
```

Comment:
The ticket only mentions this desync when unfollowing a thread. Are you sure we also need to do it when following one as well?

Reply:
My reading of the ticket suggests a forced reload of unread counts is needed for any change to the "following" status.

Reply:
I'm honestly not sure if it's required to fix this particular bug, but given that the application doesn't resync those unreads when changing the status at all, I think it's safer to do it both ways. This doesn't present a massive performance issue as manually following/unfollowing will only resync for one user.